File: 0001-0100/0053-Maximum Subarray/0053-Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum = nums[0], sum = nums[0];
        for (int i = 1; i < nums.size(); ++i) {
            if (sum < 0) {
                sum = 0;
            }
            sum += nums[i];
            maxSum = max(maxSum, sum);
        }
        
        return maxSum;
    }
};

File: 0001-0100/0053-Maximum Subarray/0053-Maximum Subarray.py
import math
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSum = -math.inf
        curr = 0
        for num in nums:
            curr += num
            maxSum = max(maxSum, curr)
            curr = max(curr, 0)
        return maxSum
